src/monattack.cpp

```diff
@@ -982,6 +1021,9 @@ void mattack::fungus_big_blossom(monster *z, int index)
 
 void mattack::fungus_inject(monster *z, int index)
 {
+    if( z->friendly ) {
+        return; // TODO: handle friendly monsters
```

Comment:
Should just ignore you and attack/make_fungus others, possibly prioritizing those adjacent to both you and it but that's not essential.